File: clang/test/CodeGenCXX/pragma-pack.cpp
// RUN: %clang_cc1 %s -triple=i686-apple-darwin10 -emit-llvm -o - | FileCheck %s

struct Base {
  virtual ~Base();
  int x;
};

#pragma pack(1)
struct Sub : virtual Base {
  char c;
};

// CHECK: %struct.Sub = type <{ ptr, i8, %struct.Base }>
Sub f(Sub x) {
  return x;
}

static int i[sizeof(Sub) == 13 ? 1 : -1];

File: clang/test/SemaObjC/pragma-pack.m
// RUN: %clang_cc1 -triple i686-apple-darwin9 -fsyntax-only -verify -Wno-objc-root-class %s
// expected-no-diagnostics

// Make sure pragma pack works inside ObjC methods.
@interface X
@end
@implementation X
- (void)Y {
#pragma pack(push, 1)
  struct x {
    char a;
    int b;
  };
#pragma pack(pop)
  typedef char check_[sizeof (struct x) == 5 ? 1 : -1];
}
@end
